frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/StrategyExecution/StrategyExecution.tsx

```diff
@@ -1,372 +1,110 @@
-    shouldForwardProp: (prop) => prop !== 'disabled',
-})<{ disabled?: boolean | null }>(({ theme, disabled }) => ({
-    '& p, & span, & h1, & h2, & h3, & h4, & h5, & h6': {
-        color: disabled ? theme.palette.neutral.main : 'inherit',
-    },
-    '.constraint-icon-container': {
-        backgroundColor: disabled
-            ? theme.palette.neutral.border
-            : theme.palette.primary.light,
-        borderRadius: '50%',
-    },
-    '.constraint-icon': {
-        fill: disabled
-            ? theme.palette.neutral.light
-            : theme.palette.common.white,
+const StyledList = styled('ul')(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    listStyle: 'none',
+    padding: 0,
+    margin: 0,
+    '&.disabled-strategy': {
+        filter: 'grayscale(1)',
+        opacity: 0.67,
     },
+    gap: theme.spacing(1),
 }));
 
-const CustomStrategyDeprecationWarning = () => (
-    <Alert severity='warning' sx={{ mb: 2 }}>
-        Custom strategies are deprecated and may be removed in a future major
-        version. Consider rewriting this strategy as a predefined strategy with{' '}
-        <Link
-            href={
-                'https://docs.getunleash.io/reference/activation-strategies#constraints'
-            }
-            target='_blank'
-            variant='body2'
-        >
-            constraints.
-        </Link>
-    </Alert>
-);
-
-const NoItems: FC = () => (
-    <Box sx={{ px: 3, color: 'text.disabled' }}>
-        This strategy does not have constraints or parameters.
-    </Box>
-);
-
-const StyledValueContainer = styled(Box)(({ theme }) => ({
+const StyledListItem = styled('li')(({ theme }) => ({
+    position: 'relative',
     padding: theme.spacing(2, 3),
     border: `1px solid ${theme.palette.divider}`,
     borderRadius: theme.shape.borderRadiusMedium,
     background: theme.palette.background.default,
 }));
 
-const StyledValueSeparator = styled('span')(({ theme }) => ({
-    color: theme.palette.neutral.main,
-}));
+const List: FC<{ children: ReactNode }> = ({ children }) => {
+    const result: ReactNode[] = [];
+    Children.forEach(children, (child, index) => {
+        if (isValidElement(child)) {
+            result.push(
+                <ListItem key={index}>
+                    {index > 0 ? (
+                        <StrategyEvaluationSeparator key={`${index}-divider`} />
+                    ) : null}
+                    {child}
+                </ListItem>,
+            );
+        }
+    });
+
+    return <StyledList>{result}</StyledList>;
+};
+
+const ListItem: FC<{ children: ReactNode }> = ({ children }) => (
+    <StyledListItem>{children}</StyledListItem>
+);
+
+type StrategyExecutionProps = {
+    strategy: IFeatureStrategyPayload | CreateFeatureStrategySchema;
+    displayGroupId?: boolean;
+};
 
-export const StrategyExecution: FC<IStrategyExecutionProps> = ({
+const NewStrategyExecution: FC<StrategyExecutionProps> = ({
```

Comment:
Probably don't want to change the name of the component, right?
```suggestion
const StrategyExecution: FC<StrategyExecutionProps> = ({
```